### ISSUE TITLE
k3s host communication broken without "entities: ["all"]" in policy

### ISSUE THREAD

**Issue by margau** [cilium/cilium#26672]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I've been trying to implement a very basic ClusterwideNetworkPolicy, allowing all cluster-internal and most external communication on a routed BGP-CP based IPv6 only k3s cluster.

As soon as the following Policy is applied, communication with the host IP breaks:

```
---
apiVersion: cilium.io/v2
kind: CiliumClusterwideNetworkPolicy
metadata:
  name: dev-cluster-policy
specs:
  - endpointSelector:
      matchLabels: {}
    ingress:
    - fromEntities:
      - host
      - remote-node
      - kube-apiserver
      - ingress
      - cluster
      - init
      - health
      - unmanaged
      - world
      #- all
    egress:
    - toEntities:
      - all
   ```
   
  result for e.g. coredns:
  ```
  Jul  6 17:19:25.116: [2001:<host pub ip>]:37134 (host) <> kube-system/coredns-77ccd57875-kflh7:8181 (ID:46852) policy-verdict:none INGRESS DENIED (TCP Flags: SYN)
Jul  6 17:19:25.116: [2001:<host pub ip>]:37134 (host) <> kube-system/coredns-77ccd57875-kflh7:8181 (ID:46852) Policy denied DROPPED (TCP Flags: SYN)
Jul  6 17:19:25.206: [2001:<host pub ip>]:57130 (host) <> kube-system/coredns-77ccd57875-kflh7:53 (ID:46852) policy-verdict:none INGRESS DENIED (UDP)
Jul  6 17:19:25.206: [2001:<host pub ip>]:57130 (host) <> kube-system/coredns-77ccd57875-kflh7:53 (ID:46852) Policy denied DROPPED (UDP)
Jul  6 17:19:25.206: [2001:<host pub ip>]:57130 (host) <> kube-system/coredns-77ccd57875-kflh7:53 (ID:46852) policy-verdict:none INGRESS DENIED (UDP)
Jul  6 17:19:25.206: [2001:<host pub ip>]:57130 (host) <> kube-system/coredns-77ccd57875-kflh7:53 (ID:46852) Policy denied DROPPED (UDP)
```

You can see the health probes and DNS requests being blocked. This happens as soon as "all" is not part of the fromEntities, even if i allow `::/0` as fromCIDR in the ingress.

Expected behavior would have been that these requests are allowed e.g. by the cluster entity, or the default route fromCIDR. The egress side shows similar behavior, thats where the "all" in egress policy comes from.

### Cilium Version

cilium-cli: v0.15.0 compiled with go1.20.4 on linux/amd64
cilium image (default): v1.13.4
cilium image (stable): v1.13.4
cilium image (running): 1.14.0-rc.0

### Kernel Version

5.10.0-23-amd64 #1 SMP Debian 5.10.179-1 (2023-05-12) x86_64 GNU/Linux

### Kubernetes Version

Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.3+k3s1", GitCommit:"fe9604cac119216e67f8bd5f14eb5608e3bcd3cf", GitTreeState:"clean", BuildDate:"2023-06-26T16:49:54Z", GoVersion:"go1.20.5", Compiler:"gc", Platform:"linux/amd64"}

### Anything else?

Ferm is deployed as host firewall, but was stopped during debugging.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by julianwiedmann** [comment]
:wave: would you be able to upload a sysdump? I think this issue could benefit from seeing the detailed cluster config.

The fact that you mention IPv6 and Host-Firewall makes me wonder why we didn't add https://github.com/cilium/cilium/pull/13049 for the IPv6 path. Does the description match your environment / symptoms?

**Comment by margau** [comment]
Hi,
sorry for the late answer.
#13049 sounds not matching at first, cause
```
kubeProxyReplacement: strict
```
is set.

Problem with a sysdump is:
This would leak the public IPs/Prefixes, which are currently not properly protected due to this bug.
Which are the most important parts, so I could publish them censored?

**Comment by margau** [comment]
Updated, it's still broken on 1.14.0.
https://github.com/cilium/cilium/issues/26886 sounds like it could be related, I'll monitor that issue too.

**Comment by margau** [comment]
Not necessarily stale, I'll try to reproduce with 1.14.2 and come back with the information.

**Comment by margau** [comment]
Seems fixed in 1.14.2
